controllers/spacerequest/spacerequest_controller.go

```diff
@@ -43,12 +63,440 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, memberClusters map[strin
+	}
+	// Add the finalizer if it is not present
+	if err := r.addFinalizer(logger, memberClusterWithSpaceRequest, spaceRequest); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	subSpace, createdOrUpdated, err := r.ensureSpace(logger, memberClusterWithSpaceRequest, spaceRequest)
+	// if there was an error or if subSpace was just created or updated,
+	// let's just return.
+	if err != nil || createdOrUpdated {
+		return ctrl.Result{}, err
+	}
+
+	// update spaceRequest conditions and target cluster url
+	err = r.updateSpaceRequest(memberClusterWithSpaceRequest, spaceRequest, subSpace)
+
+	return ctrl.Result{}, err
+}
+
+// updateSpaceRequest updates conditions and target cluster url of the spaceRequest with the ones on the Space resource
+func (r *Reconciler) updateSpaceRequest(memberClusterWithSpaceRequest cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest, subSpace *toolchainv1alpha1.Space) error {
+	// reflect Ready condition from subSpace to spaceRequest object
+	statusNeedsUpdate := reflectSubSpaceReadyCondition(subSpace, spaceRequest)
+	// set target cluster URL to space request status
+	targetClusterURL, err := r.getTargetClusterURL(subSpace.Status.TargetCluster)
+	if err != nil {
+		return err
+	}
+
+	if targetClusterURL != spaceRequest.Status.TargetClusterURL {
+		// update target cluster url if changed
+		statusNeedsUpdate = true
+		spaceRequest.Status.TargetClusterURL = targetClusterURL
+	}
+	if !statusNeedsUpdate {
+		return nil // nothing changed
+	}
+	return memberClusterWithSpaceRequest.Client.Status().Update(context.TODO(), spaceRequest)
+}
+
+// reflectSubSpaceReadyCondition reflects the type `Ready` condition from the subSpace to the spaceRequest condition list.
+// returns true if the conditions list changed
+// returns false if the conditions list is unchanged
+func reflectSubSpaceReadyCondition(subSpace *toolchainv1alpha1.Space, spaceRequest *toolchainv1alpha1.SpaceRequest) bool {
+	var addedOrUpdated bool
+	if condition.IsTrue(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady) {
+		// subSpace was provisioned so let's set Ready type condition to true on the space request
+		spaceRequest.Status.Conditions, addedOrUpdated = condition.AddOrUpdateStatusConditions(spaceRequest.Status.Conditions, toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.ConditionReady,
+			Status: v1.ConditionTrue,
+			Reason: toolchainv1alpha1.SpaceProvisionedReason,
+		})
+	} else if condition.IsFalse(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady) {
+		// subSpace is not ready let's copy the condition together with reason/message fields
+		conditionReadyFromSubSpace, _ := condition.FindConditionByType(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady)
+		spaceRequest.Status.Conditions, addedOrUpdated = condition.AddOrUpdateStatusConditions(spaceRequest.Status.Conditions, conditionReadyFromSubSpace)
+	}
+	return addedOrUpdated
+}
```

Comment:
I believe that we can simplify this
```suggestion
func (r *Reconciler) updateSpaceRequest(memberClusterWithSpaceRequest cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest, subSpace *toolchainv1alpha1.Space) error {
	// set target cluster URL to space request status
	targetClusterURL, err := r.getTargetClusterURL(subSpace.Status.TargetCluster)
	if err != nil {
		return err
	}
	spaceRequest.Status.TargetClusterURL = targetClusterURL
	// reflect Ready condition from subSpace to spaceRequest object
	if condition.IsTrue(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady) {
		// subSpace was provisioned so let's set Ready type condition to true on the space request
		return r.updateStatus(spaceRequest, memberClusterWithSpaceRequest, toolchainv1alpha1.Condition{
			Type:   toolchainv1alpha1.ConditionReady,
			Status: v1.ConditionTrue,
			Reason: toolchainv1alpha1.SpaceProvisionedReason,
		})
	} else if condition.IsFalse(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady) {
		// subSpace is not ready let's copy the condition together with reason/message fields
		conditionReadyFromSubSpace, _ := condition.FindConditionByType(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady)
		return r.updateStatus(spaceRequest, memberClusterWithSpaceRequest, conditionReadyFromSubSpace)
	}
	return nil
}
```
the point is that it won't ever happen that we would update SR's conditions to the "ready state" without having the target cluster set in Space.status (and thus also in SpaceRequest.status). WDYT?

Reply:
the second option would be to reshuffle the code a bit - to make it easier to read
```suggestion
func (r *Reconciler) updateSpaceRequest(memberClusterWithSpaceRequest cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest, subSpace *toolchainv1alpha1.Space) error {
	// set target cluster URL to space request status
	targetClusterURL, err := r.getTargetClusterURL(subSpace.Status.TargetCluster)
	if err != nil {
		return err
	}
	
	readyCondition, _ := condition.FindConditionByType(spaceRequest.Status.Conditions, toolchainv1alpha1.ConditionReady) 
	// reflect Ready condition from subSpace to spaceRequest object
	if condition.IsTrue(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady) {
		// subSpace was provisioned so let's set Ready type condition to true on the space request
		readyCondition = toolchainv1alpha1.Condition{
			Type:   toolchainv1alpha1.ConditionReady,
			Status: v1.ConditionTrue,
			Reason: toolchainv1alpha1.SpaceProvisionedReason,
		}
	} else if condition.IsFalse(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady) {
		// subSpace is not ready let's copy the condition together with reason/message fields
		readyCondition, _ = condition.FindConditionByType(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady)
	}
	
	var updated bool
	spaceRequest.Status.Conditions, updated = condition.AddOrUpdateStatusConditions(spaceRequest.Status.Conditions, readyCondition)
	if !updated && spaceRequest.Status.TargetClusterURL == targetClusterURL {
		// Nothing changed
		return nil
	}
	spaceRequest.Status.TargetClusterURL = targetClusterURL
	return memberClusterWithSpaceRequest.Client.Status().Update(context.TODO(), spaceRequest)
}
```
this is maybe better and safer option

Reply:
I ended up adding your first suggestion in [60159bc](https://github.com/codeready-toolchain/host-operator/pull/770/commits/60159bc49fc4872acb0b880853cb3775e5454300)

Because, If I'm not wrong, the second suggestion may create a `panic` since  `condition.AddOrUpdateStatusConditions(spaceRequest.Status.Conditions, readyCondition)` may end up being called with and "empty" condition object `toolchainv1alpha1.Condition{}` and a function in that stack call tries to access directly `newCondition.Type` (which will be nil) https://github.com/codeready-toolchain/toolchain-common/blob/7d686b51beafc77a3db03a4ee66671582eb2b19c/pkg/condition/condition.go#L120.

If I'm not missing anything, I think your first suggestion it's safer. WDYT?